x/delayedack/ibc_middleware.go

```diff
@@ -148,13 +158,29 @@ func (im IBCMiddleware) OnRecvPacket(
 	rollappPacket := types.RollappPacket{
 		RollappId:   chainID,
 		Packet:      &packet,
-		Status:      types.RollappPacket_PENDING,
+		Status:      commontypes.Status_PENDING,
 		Relayer:     relayer,
 		ProofHeight: ibcClientLatestHeight.GetRevisionHeight(),
 		Type:        types.RollappPacket_ON_RECV,
 	}
 	im.keeper.SetRollappPacket(ctx, rollappPacket)
 
+	// Handle eibc demand order if exists
```

Comment:
I would move this snippet and `createDemandOrderFromIBCPacket` to a seperate file